src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -479,88 +576,260 @@ private async Task<ProjectChanges> GetProjectChangesAsync(Project project, Cance
+            }
+
+            return ImmutableArray<LocationlessDiagnostic>.Empty;
+        }
+
+        public async Task<SolutionUpdate> EmitSolutionUpdateAsync(Solution solution, CancellationToken cancellationToken)
         {
+            if (StoppedAtException)
+            {
+                return SolutionUpdate.Blocked();
+            }
+
+            var deltas = ArrayBuilder<Deltas>.GetInstance();
+            var emitBaselines = ArrayBuilder<(ProjectId, EmitBaseline)>.GetInstance();
+            var deferredDeltas = ArrayBuilder<Deltas>.GetInstance();
+            var moduleReaders = ArrayBuilder<IDisposable>.GetInstance();
+            var diagnostics = ArrayBuilder<(ProjectId, ImmutableArray<Diagnostic>)>.GetInstance();
+
             try
             {
-                Debug.Assert(!_stoppedAtException);
+                // TODO: parallelize
 
-                var projectChanges = await GetProjectChangesAsync(project, cancellationToken).ConfigureAwait(false);
-                var currentCompilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-                var allAddedSymbols = await GetAllAddedSymbolsAsync(project, cancellationToken).ConfigureAwait(false);
-                var baseActiveStatements = await BaseActiveStatements.GetValueAsync(cancellationToken).ConfigureAwait(false);
-                var baseActiveExceptionRegions = await BaseActiveExceptionRegions.GetValueAsync(cancellationToken).ConfigureAwait(false);
+                bool isBlocked = false;
+
+                foreach (var project in solution.Projects)
+                {
+                    if (!SupportsEditAndContinue(project))
+                    {
+                        continue;
+                    }
+
+                    var baseProject = GetBaseProject(project.Id);
+
+                    // TODO (https://github.com/dotnet/roslyn/issues/1204):
+                    // When debugging session is started some projects might not have been loaded to the workspace yet. 
+                    // We capture the base solution. Edits in files that are in projects that haven't been loaded won't be applied
+                    // and will result in source mismatch when the user steps into them.
+                    // TODO: hook up the debugger reported error, check that the project has not been loaded and report a better error.
+                    // Here, we assume these projects are not modified.
+                    if (baseProject == null)
+                    {
+                        EditAndContinueWorkspaceService.Log.Write("Emitting update of '{0}' [0x{1:X8}]: project not loaded", project.Id.DebugName, project.Id);
+                        continue;
+                    }
+
+                    var (mvid, mvidReadError) = await DebuggingSession.GetProjectModuleIdAsync(project, cancellationToken).ConfigureAwait(false);
```

Comment:
❗️ Needs to use `ConfigureAwait(true)` since this code follows a JTF context switch in the execution flow (per the containing `foreach`).

```suggestion
                    var (mvid, mvidReadError) = await DebuggingSession.GetProjectModuleIdAsync(project, cancellationToken).ConfigureAwait(true);
```